core/src/main/java/org/apache/calcite/sql/SqlFunction.java

```diff
@@ -271,18 +271,20 @@ private RelDataType deriveType(
         return validator.deriveConstructorType(scope, call, this, function,
             argTypes);
       }
-      if (function == null && validator.isTypeCoercionEnabled()) {
+      if (function == null) {
         // try again if implicit type coercion is allowed.
```

Comment:
better to move this doc inside ` if (validator.isTypeCoercionEnabled()) {` ?

Reply:
I agree, thanks.